edb/lang/edgeql/compiler/viewgen.py

```diff
@@ -173,11 +173,18 @@ def _normalize_view_ptr_expr(
     qlexpr = None
 
     if is_polymorphic:
-        source = qlast.TypeFilter(
-            expr=qlast.Path(steps=[qlast.Source()]),
-            type=qlast.TypeName(maintype=steps[0]))
+        ptype = steps[0]
+        # FIXME: what are the options here?
+        if isinstance(ptype, qlast.TypeName):
```

Comment:
Wouldn't it be `TypeExpr` here?

Reply:
Yeah. I'm also not sure there's an `if..else` situation there. A polymorphic query *would* have a TypeExpr and that's that. So I'll use an `isinstance` test instead of a `len` test.